modules/platforms/python/pyignite/cache.py

```diff
@@ -0,0 +1,592 @@
+         should be converted,
+        :return: value retrieved.
+        """
+        result = cache_get(self._client, self._cache_id, key, key_hint=key_hint)
+        result.value = self._process_binary(result.value)
+        return result
+
+    @status_to_exception(CacheError)
+    def put(self, key, value, key_hint: object=None, value_hint: object=None):
+        """
+        Puts a value with a given key to cache (overwriting existing value
+        if any).
+
+        :param key: key for the cache entry. Can be of any supported type,
+        :param value: value for the key,
+        :param key_hint: (optional) Ignite data type, for which the given key
+         should be converted,
+        :param value_hint: (optional) Ignite data type, for which the given
+         value should be converted.
+        """
+        return cache_put(
+            self._client, self._cache_id, key, value,
+            key_hint=key_hint, value_hint=value_hint
+        )
+
+    @status_to_exception(CacheError)
+    def get_all(self, keys: list) -> list:
+        """
+        Retrieves multiple key-value pairs from cache.
+
+        :param keys: list of keys or tuples of (key, key_hint),
+        :return: a dict of key-value pairs.
+        """
+        result = cache_get_all(self._client, self._cache_id, keys)
+        if result.value:
+            for key, value in result.value.items():
+                result.value[key] = self._process_binary(value)
+        return result
+
+    @status_to_exception(CacheError)
+    def put_all(self, pairs: dict):
+        """
+        Puts multiple key-value pairs to cache (overwriting existing
+        associations if any).
+
+        :param pairs: dictionary type parameters, contains key-value pairs
+         to save. Each key or value can be an item of representable
+         Python type or a tuple of (item, hint),
+        """
+        return cache_put_all(self._client, self._cache_id, pairs)
+
+    @status_to_exception(CacheError)
+    def replace(
+        self, key, value, key_hint: object=None, value_hint: object=None
+    ):
+        """
+        Puts a value with a given key to cache only if the key already exist.
+
+        :param key: key for the cache entry. Can be of any supported type,
+        :param value: value for the key,
+        :param key_hint: (optional) Ignite data type, for which the given key
+         should be converted,
+        :param value_hint: (optional) Ignite data type, for which the given
+         value should be converted.
+        """
+        result = cache_replace(
+            self._client, self._cache_id, key, value,
+            key_hint=key_hint, value_hint=value_hint
+        )
+        result.value = self._process_binary(result.value)
+        return result
+
+    @status_to_exception(CacheError)
+    def clear(self, keys: Optional[list]=None):
+        """
+        Clears the cache without notifying listeners or cache writers.
```

Comment:
Removing values from cache without notifying listeners or cache writers more looks like an security issue than correct behaviour.

Reply:
Looks like this is JCache specified behaviour

Reply:
Could you please elaborate on what might the security threat be? This operation is not directly related to authentication or encryption.